File: 550.MaximumDepthofN-aryTree.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    int maxDepth(Node* root) {
        if(root == nullptr) return 0;
        
        queue<Node*> q;
        q.push(root);
        
        int depth = 0;
        
        while(!q.empty()) {
            depth++;
            int qSize = q.size();
            
            for(int i = 0; i < qSize; i++) {
                Node* node = q.front();
                q.pop();
                for(const auto& child : node->children) {
                    q.push(child);
                }
            }
        }
        
        return depth;
    }
};

File: 550.MaximumDepthofN-aryTree.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""
import queue

class Solution:
    def maxDepth(self, root: 'Node') -> int:
        if root == None:
            return 0
        data_queue = queue.Queue()
        data_queue.put(root)
        
        depth = 0
        
        while not data_queue.empty():
            depth = depth + 1
            queue_size = data_queue.qsize()
            for i in range(queue_size):
                node = data_queue.get()
                for child in node.children:
                    data_queue.put(child)
        
        return depth